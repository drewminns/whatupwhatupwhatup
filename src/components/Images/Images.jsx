import React, { Component } from 'react';
import { connect } from 'react-redux';
import classNames from 'classnames';
import { updateState } from '../../actions';

class Images extends Component {

  componentWillMount() {
    this.props.updateState();
  }

  render() {
    return (
      <div
        className={classNames('foo', 'bar')}
      >
        Hello
      </div>
    );
  }
}

Images.propTypes = {
  updateState: React.PropTypes.func.isRequired,
};

const mapDispatchToProps = dispatch => ({
  updateState: val => dispatch(updateState(val)),
});

const mapStateToProps = state => ({
  app: state.app,
});

export default connect(mapStateToProps, mapDispatchToProps)(Images);
