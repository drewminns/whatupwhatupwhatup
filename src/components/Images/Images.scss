$cool: green;
.images {
  background: $cool;
}
