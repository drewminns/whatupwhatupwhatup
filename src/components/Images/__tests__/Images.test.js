import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Images from '../Images';

describe('Images component renders correctly', () => {
  const text = 'Hello testie McTest';

  it('renders correctly', () => {
    const rendered = renderer.create(
      <Images text={text}/>
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });

  it('renders in enzyme', () => {
    const ImagesComponent = shallow(<Images text={text} />);
    expect(ImagesComponent.text()).toEqual(text);
  });
});
