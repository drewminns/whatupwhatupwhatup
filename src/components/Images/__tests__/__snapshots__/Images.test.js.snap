exports[`Images component renders correctly renders correctly 1`] = `
<div
  className="images">
  <p>
    Hello testie McTest
  </p>
</div>
`;
